service/build.gradle

```diff
@@ -20,249 +20,30 @@ plugins {
-    useJUnitPlatform {
-        includeTags "unit"
-    }
-    testLogging {
-        events = ["passed", "failed", "skipped", "started"]
-    }
-    outputs.upToDateWhen { false }
-    finalizedBy jacocoTestReport
-}
-
-task connectedTest(type: Test) {
-    environment.put('TEST_ENV', System.getenv("TEST_ENV"))
-    useJUnitPlatform {
-        includeTags "connected"
-    }
-    testLogging {
-        events = ["passed", "failed", "skipped", "started"]
-    }
-    outputs.upToDateWhen { false }
-}
-
-
-task integrationTest(type: Test) {
-    environment.put('TEST_ENV', System.getenv("TEST_ENV"))
-    useJUnitPlatform {
-        includeTags "integration"
-    }
-    testLogging {
-        events = ["passed", "failed", "skipped", "started"]
-    }
-    outputs.upToDateWhen { false }
-}
-
-spotless {
-    java {
-        googleJavaFormat()
-        target 'src/*/java/**/*.java'
-    }
-}
-
-// Generate/clean the version properties file
-// This was being done by a plugin: https://github.com/palantir/gradle-git-version
-// but it does not work with Gradle 7.0. It was also very complicated for a
-// what we need, so easily replaced with a bash script
-def generatedVersionFile = "${resourceDir}/generated/version.properties"
-
-task generateVersionProperties(type: Exec) {
-    commandLine "${projectDir}/writeVersionProperties.sh", "${version}", "${generatedVersionFile}"
+project.ext {
+    artifactGroup = "${group}.workspace"
+    openapiOutputDir = "${buildDir}/openapi"
+    resourceDir = "${projectDir}/src/main/resources"
 }
 
-task cleanVersionProperties(type: Exec) {
-    commandLine "rm", "-f", "${generatedVersionFile}"
-}
-
-// Deploy config
-jib {
-    from {
-        // see https://github.com/broadinstitute/dsp-appsec-blessed-images/tree/main/jre
-        // Google's distroless images are openjdk, this is the simplest with adoptopenjdk
-        image = "us.gcr.io/broad-dsp-gcr-public/base/jre:11-debian"
-    }
-    extraDirectories {
-        paths = [file(jibExtraDirectory)]
-    }
-    container {
-        filesModificationTime = java.time.ZonedDateTime.now().toString() // to prevent ui caching
-        jvmFlags = [
-                "-agentpath:" + cloudProfilerLocation + "/profiler_java_agent.so=" +
-                        "-cprof_service=bio.terra.workspace" +
-                        ",-cprof_service_version=" + version +
-                        ",-cprof_enable_heap_sampling=true" +
-                        ",-logtostderr" +
-                        ",-minloglevel=2"
-        ]
-    }
-}
+def includeDir = "$projectDir/gradle"
+// include order matters, so don't alphabetize (yet)
+apply(from: "$includeDir/buildScan.gradle")
+apply(from: "$includeDir/profiler.gradle")
+apply(from: "$includeDir/testing.gradle")
+apply(from: "$includeDir/deploy.gradle")
+apply(from: "$includeDir/lint.gradle")
+apply(from: "$includeDir/openapi.gradle")
+apply(from: "$includeDir/versionProperties.gradle")
+apply(from: "$includeDir/dependencies.gradle")
```

Comment:
I'm not loving this. Is the file really so big that it needs to be exploded into 8 pieces? That won't make it easier for me anyway.

I wouldn't mind factoring out parts that are common. Like the spotless invocation and the jacoco and the buildscan are boilerplate.

The dependencies and tests are not.

Reply:
There's lots of room to massage this. @gjuggler has done some similar grouping in the common lib.

It's not about the size of the file so much as the number of different concerns that are in it. It's bewildering to me to see so many different aspects of the build all in one file. If it were a Java class, we'd surely split it up based on the single responsibility principle.